Assembly Workshop Week 6: Wrap up
---

Tutorial steps:  
1. Visualize Kraken results with Krona  

###1. Visualize Kraken results with Krona
* [Krona](https://github.com/marbl/Krona/wiki) is a hierarchical data visualization tool that works nicely for taxonomies.
* Remember that Kraken will classify only reads or contigs (whichever kind of input data you chose) with exact kmer matches to the Bacteria, Archaea, Virus RefSeq database. Reads or contigs without matches are presumed to be uncontaminated (at least not by Bacteria, Archaea, or Viruses).
* Download Krona onto your computer from [here.](https://github.com/marbl/Krona/wiki/Downloads)
* You will need to grab the log file generated by ```kraken-translate``` (the one with taxon names from Kraken's second step). Make sure that when you run ```kraken-translate``` to include the flag ```--mpa-format```.
	+ If you don't have Kraken results, copy this one to your working directory:  
```/pool/genomics/dikowr/assembly_tutorial/kraken_results/Lipotes_vexillifer_names.log```

* We'll need to adjust the Kraken output to be able to be read by Krona. I have a python script up on the github that will do this. 
	+ [script:](https://github.com/SmithsonianWorkshops/GenomeAssembly/blob/master/rename_Kraken_output_for_Krona.py) ```rename_Kraken_output_for_Krona.py```
	+ ```python rename_Kraken_output_for_Krona.py <your_kraken_output> > <your_desired_krona_input>```  
* Then you can run the ```ImportRDP.pl``` script within Krona to generate a html file you can interact with in a web browser (the scripts are in a directory called ```scripts```). RDP stands for the RDP classifier, which is often used in QIIME. Our Kraken output resembles the RDP output, so this script adjusts the Kraken output to be the same as the RDP output and thus readable by Krona.
	+ ```perl ImportRDP.pl <your_desired_krona_input> -o <your_output.html>```
	+ Click on the html file to open in a browser!